#855 - Fixing CSharpScriptEngine script loading

abosma opened the pull request:
As seen in issue https://github.com/LeagueSandbox/GameServer/issues/855 only scripts from the latest loaded package are loaded in, this is due to the Assembly being overwritten by the latest package.

Every loaded package will now have it's own assembly loaded in, this means every script can now be loaded in instead of only scripts in the last loaded package.

abosma committed:
Fixed CSharpScriptEngine only loading in scripts from last loaded package

microsoftv commented on GameServerLib/Scripting/CSharp/CSharpScriptEngine.cs:
```diff
@@ -115,32 +115,54 @@ private List<SyntaxTree> GetInvalidSourceTrees(EmitResult result)
             {
-                _logger.Warn($"Failed to load script: {scriptNamespace}.{scriptClass}");
-                return default(T);
+                var classType = scriptAssembly.GetType(scriptNamespace + "." + scriptClass);
```
Results in an error.

abosma commented:
Updated all Nuget packages to their latest versions (This is done since scripts can not use certain newer code functions because of the extremely outdated packages used by CSharpScriptEngine). Changed AppVeyor.yml by removing GameServerApp which was always skipped, and by changing the version to the new version of OpenCover. And removed unused references from all projects to clean up everything.

abosma commented:
Removed Nuget updating, shouldn't be in this PR.

Veykril commented on GameServerLib/Scripting/CSharp/CSharpScriptEngine.cs:
```diff
@@ -115,32 +115,54 @@ private List<SyntaxTree> GetInvalidSourceTrees(EmitResult result)
         public T GetStaticMethod<T>(string scriptNamespace, string scriptClass, string scriptFunction)
         {
-            if (_scriptAssembly == null) return default(T);
+            if (_scriptAssembly == null || !_scriptAssembly.Any())
```
Would prefer to have `!_scriptAssembly.Any()` replaced by `_scriptAssembly.Count <= 0`, will make the intent more clear i think

Veykril commented on GameServerLib/Scripting/CSharp/CSharpScriptEngine.cs:
```diff
@@ -115,32 +115,54 @@ private List<SyntaxTree> GetInvalidSourceTrees(EmitResult result)
 
         public T CreateObject<T>(string scriptNamespace, string scriptClass)
         {
+            if (_scriptAssembly == null || !_scriptAssembly.Any())
```
same here

abosma committed:
worlds tiniest refactor

danil179 commented on GameServerLib/Scripting/CSharp/CSharpScriptEngine.cs:
```diff
@@ -115,7 +115,7 @@ private List<SyntaxTree> GetInvalidSourceTrees(EmitResult result)
         public T GetStaticMethod<T>(string scriptNamespace, string scriptClass, string scriptFunction)
         {
-            if (_scriptAssembly == null || !_scriptAssembly.Any())
+            if (_scriptAssembly == null || _scriptAssembly.Count > 0)
```
Should be `<=0`

abosma committed:
worlds tiniest fix

danil179 reviewed (changes requested):
See comment above.

danil179 reviewed (approved):
Approved.
